.github/workflows/versionincrement.yaml

```diff
@@ -0,0 +1,77 @@
+---
+name: Version Increment
+on:
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: Log level
+        required: true
+        default: warning
+        type: choice
+        options:
+          - info
+          - warning
+          - debug
+
+jobs:
+  plugin-version-increment-sync:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # repo: [common-utils, job-scheduler, ml-commons, performance-analyzer, security, alerting, index-management, sql, observability, asynchronous-search, k-NN, anomaly-detection, cross-cluster-replication, dashboards-reports]
+        # repo: [job-scheduler, common-utils]
+        repo: [anomaly-detection]
+        branch: [{opensearch_branch: 1.3, plugin_branch: 1.3}]
+      # exclude:
+      # - {repo: notifications, branch: {opensearch_branch: "1.x", plugin_branch: "1.x"}}
+      # - {repo: notifications, branch: {opensearch_branch: "1.3", plugin_branch: "1.3"}}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Check out OpenSearch repo
+        uses: actions/checkout@v3
+        with:
+          repository: opensearch-project/OpenSearch
+          ref: ${{ matrix.branch.opensearch_branch }}
+      - name: Fetch OpenSearch version
+        run: |
+          OPENSEARCH_VERSION=$(cat buildSrc/version.properties | grep opensearch | cut -d= -f2 | grep -oE '[0-9.]+')
+          OPENSEARCH_VERSION=$OPENSEARCH_VERSION-SNAPSHOT
+          echo "OPENSEARCH_VERSION=$OPENSEARCH_VERSION" >> $GITHUB_ENV
+          echo "OPENSEARCH_VERSION_NUMBER=$OPENSEARCH_VERSION" | cut -f1 -d"-" >> $GITHUB_ENV
+      - name: Check out plugin repo
+        uses: actions/checkout@v3
+        with:
+          repository: prudhvigodithi/${{ matrix.repo }}
+          ref: ${{ matrix.branch.plugin_branch }}
+      - name: version comparision
+        run: |
+          OPENSEARCH_EXISTING_VERSION=$(grep opensearch.version gradle.properties | cut -d'=' -f 2-)
+          echo "OPENSEARCH_EXISTING_VERSION=$OPENSEARCH_EXISTING_VERSION" >> $GITHUB_ENV
+          echo "OPENSEARCH_EXISTING_VERSION_NUMBER=$OPENSEARCH_EXISTING_VERSION" | cut -f1 -d"-" >> $GITHUB_ENV
+      - name: Version Increment plugin repo
+        if: env.OPENSEARCH_EXISTING_VERSION_NUMBER < env.OPENSEARCH_VERSION_NUMBER
```

Comment:
I don't think you can compare version numbers like this. A string `1.10.2` will be smaller than `1.9.0`. I do think a "different", aka `!=` works here.

Reply:
Thanks @dblock for the inputs, `!=` looks better fit, as there are very less chances a plugin gets incremented before core. Let me test the workflow with `!=`.
Thank you

Reply:
`!=` works @dblock I have tested the workflow
https://github.com/prudhvigodithi/opensearch-build/runs/7195559202?check_suite_focus=true